### ISSUE TITLE
[$500]  IOU - "Out of pocket" and "Company spend" appear in IOU report when changing requested amount

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#33617]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.17-1
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
1. [User A] Create two money requests from User B
2. [User A] Stay in IOU report
3. [User B] Go to IOU report > + > Request money
4. [User B] Create a money request from User A which is lower than what's owed
5. [User B] Go to IOU details page of request created in Step 4
6. [User B] Change the amount to higher amount (but still lower than what's owed)
7. [User B] Return to IOU report
8. [User B] Repeat Step 6 if "Out of pocket" and "Company spend" do not appear in IOU report for User B
9. [User B] Pay the amount

## Expected Result:
In Step 7, there should be no "Out of pocket" and "Company spend" in IOU report of both users

## Actual Result:
In Step 7, "Out of pocket" and "Company spend" appear in IOU report of both users

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/1c098141-c566-462c-9058-4f93fd69bf38

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0143f145e728e7ea82</li>
        <li>Upwork Job ID: 1739731865357979648</li>
        <li>Last Price Increase: 2024-01-16</li>
    </ul>
</details>

**Comment by rrrshtt** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

### What is the root cause of that problem?
https://github.com/Expensify/App/blob/6365aebb9cdb3dd4a66e77117afd0fb1b1fa8957/src/libs/ReportUtils.ts#L1652
There is a problem in this function. We should actually check "reimbursable" in the transaction to ensure that some money is indeed "reimbursable." I don't completely understand how the amount of nonReimbursable is calculated, as there is zero code for modifying it.

### What changes do you think we should make in order to solve the problem?

Verify information about transactions to ensure that we have marked those transactions as reimbursable. Only after confirming their reimbursable status, the function should proceed to calculate the values.

### What alternative solutions did you explore? (Optional)
In `getUpdateMoneyRequestParams()`, we have `TransactionUtils.getUpdatedTransaction()` and `ReportUtils.getTransactionDetails()` with zero information about the "reimbursable" property.

**Comment by rrrshtt** [comment]
> > There is a problem in this function. We should actually check "reimbursable" in the transaction to ensure that some money is indeed "reimbursable." I don't completely understand how the amount of nonReimbursable is calculated, as there is zero code for modifying it.
> 
> It might be possible that these values are coming from the backend.

Take a look at the route handler. We need to check why, when updating the previous value to a higher value, the difference is calculated as "Company spend."
It's my first day of familiarization with the codebase, and I'm not sure if I have access to look at the backend side.

**Comment by rrrshtt** [comment]
@parasharrajat I updated my proposal, problem on backend-side. Second part of problem in this issue explained in comment on another issue. I can fix it here if needed.

**Comment by bfitzexpensify** [comment]
Rather than have this issue focus on two different things, let's keep using https://github.com/Expensify/App/issues/33816 for the copying issue. Updated the OP to remove that part. 

@parasharrajat do you agree that the problem here with the reimbursable/non-reimbursable calculations and flow-on effects are a BE issue?

**Comment by bfitzexpensify** [comment]
Bump on q in https://github.com/Expensify/App/issues/33617#issuecomment-1879252108 @parasharrajat, thank you

**Comment by bfitzexpensify** [comment]
Bumping the q in https://github.com/Expensify/App/issues/33617#issuecomment-1879252108 again @parasharrajat

**Comment by bfitzexpensify** [comment]
I'm going to be mostly ooo until Jan 29, assigning a second BZ team member to keep an eye on this.

- @parasharrajat is reviewing the suggestion in https://github.com/Expensify/App/issues/33617#issuecomment-1869851632 that this is a BE issue
- if it is BE, I think we could suggest this belong in the #vip-split project and then try and recruit an internal volunteer

**Comment by puneetlath** [comment]
@parasharrajat would you mind summarizing the status of this issue?

**Comment by FitseTLT** [comment]
I think it is already solved in BE.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by puneetlath** [comment]
Ok going to go ahead and close this then. @bfitzexpensify or anyone else, feel free to comment or reopen if that's the wrong thing to do.
